### ISSUE TITLE
ContainerCluster SecretKeyRef wrong type

### ISSUE THREAD

**Issue by aa250816** [GoogleCloudPlatform/k8s-config-connector#468]
## Checklist

- [ ] I did not find a related open issue.
- [ ] I did not find a solution in the troubleshooting guide: (https://cloud.google.com/config-connector/docs/troubleshooting)
- [ ] If this issue is time-sensitive, I have submitted a corresponding issue with [GCP support](https://cloud.google.com/support-hub).

## Bug Description

The go types for a container cluster SecretKeyRef has the type v1alpha1.ResourceRef [link.](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/pkg/apis/container/v1beta1/containercluster_types.go#L556) ResourceRef has the following fields
```
/* The external name of the referenced resource */
	External string `json:"external,omitempty"`
	/* Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names */
	Name string `json:"name,omitempty"`
	/* Namespace of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/ */
	Namespace string `json:"namespace,omitempty"`
```

The crd for containercluster expects SecretKeyRef to have the fields name and key. Using the go types we can get or set the key value since the field doesn't exist. 

### Config Connector Version

1.49.0

### Config Connector Mode

cluster

## Steps to Reproduce

Use go types to try and and create a cluster password
```cluster.Spec.MasterAuth.Password = &clusterv1beta1.ClusterPassword{
		ValueFrom: &clusterv1beta1.ClusterValueFrom{
			SecretKeyRef: &kcc.ResourceRef{
				Name: BasicAuthSecretName(cluster.Name),
				Key:  PasswordKey,
			},
		},
	}```

**Comment by maqiuyujoyce** [comment]
Hi @aa250816 , thank you for bringing up this issue. We'll look into it and will keep you updated.

**Comment by maqiuyujoyce** [comment]
Hi @aa250816 , could you share some more information about the priority/importance of this bug? Is it blocking, a friction-point, or nice-to-have?

**Comment by aa250816** [comment]
We have our own go structs we've made before the official ones we're added. It's not blocking but it is preventing us from using the official ones in our project.

**Comment by xiaobaitusi** [comment]
Hi @aa250816, thanks for the clarification. Glad to hear that you have a workaround to unblock yourself, we will work on the fix and keep this issue updated.
